js/app.js

```diff
@@ -118,15 +118,30 @@ TextApp.prototype.initEditor_ = function() {
   // set up all dependent objects
   this.dialogController_ = new DialogController($('#dialog-container'),
                                                 this.editor_);
-  this.tabs_ = new Tabs(this.editor_, this.dialogController_, this.settings_);
+  if (!this.tabs_) {
+    this.tabs_ = new Tabs(this.editor_, this.dialogController_, this.settings_);
+    this.menuController_ = new MenuController(this.tabs_);
+  } else {
+    // if tabs already exists, just replace the editor so we preserve state
+    this.tabs_.updateEditor(this.editor_);
+  }
 
```

Comment:
nit: delete blank lines so that all the dependent object setup code is grouped under the code comment 'set up all dependent objects' above